src/components/ChainSelector/ChainSelector.styled.tsx

```diff
@@ -37,76 +36,58 @@ const CommonButtonStyle = styled(Button)(() => {
   }
 })
 
-const SelectorButton = styled(CommonButtonStyle)(() => {
-  const theme = useTheme()
-  return {
-    [theme.breakpoints.down("sm")]: {
-      "&.MuiButton-sizeMedium.MuiButton-containedSecondary": {
-        paddingLeft: "0",
-        paddingRight: "0",
-        minWidth: "50px",
-      },
-      "& span": {
-        marginLeft: "0",
-        marginRight: "0",
-      },
+const SelectorButton = styled(CommonButtonStyle)(({ theme }) => ({
+  [theme.breakpoints.down("sm")]: {
+    "&.MuiButton-sizeMedium.MuiButton-containedSecondary": {
+      paddingLeft: "0",
+      paddingRight: "0",
+      minWidth: "50px",
     },
-  }
-})
-
-const ChainButton = styled(CommonButtonStyle)((props: {
-  isSelected: boolean
-}) => {
-  const { isSelected } = props
-  const theme = useTheme()
-
-  let background = "transparent"
-
-  if (isSelected) {
-    background = theme.palette.action.hover
-  }
+    "& span": {
+      marginLeft: "0",
+      marginRight: "0",
+    },
+  },
+}))
 
+const ChainButton = styled(Button)<ChainButtonProps>((props) => {
+  const { isActive, isDisabled, theme } = props
   return {
+    display: "flex",
     alignItems: "center",
     justifyContent: "flex-start",
-    marginBottom: "8px",
-    "&.MuiButton-sizeMedium.MuiButton-textSecondary:not(.Mui-disabled):not(.Mui-focusVisible):not(:hover)":
-      {
-        color: theme.palette.text.primary,
-        fontSize: "17px",
-        textTransform: "capitalize" as const,
-        backgroundColor: background,
-      },
-    "&.MuiButton-sizeMedium.MuiButton-textSecondary:not(.Mui-disabled):not(.Mui-focusVisible):hover":
-      {
-        color: theme.palette.text.primary,
-        fontSize: "17px",
-        textTransform: "capitalize" as const,
-      },
+    width: "100%",
+    padding: "12px 16px",
+    backgroundColor: isActive ? theme!.palette.primary.dark : "transparent",
+    color: isActive
+      ? theme!.palette.primary.contrastText
+      : theme!.palette.text.primary,
+    opacity: isDisabled ? 0.5 : 1,
+    "&:hover": {
+      backgroundColor: isActive
+        ? theme!.palette.primary.dark
+        : theme!.palette.action.hover,
```

Comment:
Why do we need to enforce that the them is defined?

Reply:
We need to put it as optional, because it'll be injected by `styled` function, but when we use it as component there is no theme.